Controller/Gateway/MoneroPayment.php

```diff
@@ -399,120 +474,145 @@ public function execute()
         $country = '';
         $region = '';
         $paramCount = 0;
-        
-        if(isset($_GET['first'])){
-            $first = $_GET['first'];
+        $errors = [];
+
+        if (isset($_GET['first'])) {
+            $first = htmlspecialchars(filter_input(INPUT_GET, 'first'), ENT_QUOTES, 'UTF-8');
+            if (empty($first)) {
+                $errors[] = 'First name cannot be empty.';
+            }
             $paramCount += 1;
         }
-        if(isset($_GET['last'])){
-            $last = $_GET['last'];
+        if (isset($_GET['last'])) {
```

Comment:
Better `last` ---> `last_name`

Reply:
It is being mapped to last name in $orderData: `'lastname' => $last,`